presidio-analyzer/presidio_analyzer/predefined_recognizers/phone_recognizer.py

```diff
@@ -66,51 +54,34 @@ def analyze(
         :return: List of phone numbers RecognizerResults
         """
         results = []
-        for entity in entities:
-            region = entity.replace(ENTITY_TYPE_SUFFIX, "")
-            if region in SUPPORTED_REGIONS or entity is INTERNATIONAL_ENTITY_TYPE:
-                for match in phonenumbers.PhoneNumberMatcher(text, region, leniency=0):
-                    international_phone_prefix = match.raw_string.startswith("+")
-                    if entity == INTERNATIONAL_ENTITY_TYPE \
-                            and international_phone_prefix:
-                        results += [self._get_international_recognizer_result(match)]
-                    # phone-numbers matches international numbers twice
-                    elif not international_phone_prefix:
-                        results += [
-                            self._get_regional_recognizer_result(
-                                match, entity, text, nlp_artifacts
-                            )
-                        ]
-
-        return results
-
-    def _get_regional_recognizer_result(self, match, entity, text, nlp_artifacts):
+        for region in self.supported_regions:
+            for match in phonenumbers.PhoneNumberMatcher(text, region, leniency=0):
+                results += [self._get_recognizer_result(match, text, region, nlp_artifacts)]
+
+        return EntityRecognizer.remove_duplicates(results)
```

Comment:
Eventually the implementation of this method didn't allow easy aggregation of the analysis explanation to include all matched regions. Decided to not include regions in the explanation.